doc/content/getting-started/migrating-from-v2/major-changes.md

```diff
@@ -0,0 +1,83 @@
+
+With Packet Broker enabled, you can receive traffic on {{% tts %}} without the need to configure any of your gateways.
+
+For private {{% tts %}} deployments with Packet Broker disabled, you will need to re-configure your gateways to connect to {{% tts %}}, so that you can start receiving traffic from your end devices.
+
+In order to connect a gateway to {{% tts %}}, follow instructions for [Adding a Gateway in the Console]({{< ref "/getting-started/console/create-gateway" >}}) or [Adding a Gateway using the CLI]({{< ref "/getting-started/cli#create-gateway" >}}). Then, reconfigure the gateway to connect to {{% tts %}}, and regenerate its API key (if required).
+
+## Application Data
+
+{{% tts %}} uses a different data format for uplink and downlink traffic than {{% ttnv2 %}}.
+
+Example uplink message on {{% ttnv2 %}}:
+
+```js
+/* topic name: app1/devices/dev1/up */
+{"app_id":"app1","dev_id":"dev1","hardware_serial":"1122334411223344","port":1,"counter":0,"payload_raw":"EQ==","payload_fields":{"led":17},"metadata":{"time":"2020-05-01T00:04:41.258830149Z","latitude":47.984,"longitude":43.123,"altitude":100}}
+```
+
+Example downlink message on {{% ttnv2 %}}:
+
+```js
+/* topic name: app1/devices/dev1/down */
+{"port":10,"confirmed":true,"payload_raw":"EBA="}
+```
+
+The data format has changed in {{% tts %}}. It uses a different schema, different names, and has much richer metadata support. Read more about that in [Data Formats]({{% ref "/integrations/data-formats" %}}) and in [Working With Data]({{% ref "/getting-started/working-with-data" %}}).
+
+When migrating to {{% tts %}}, ensure your application can properly handle the new {{% tts %}} data format.
+
+### MQTT Traffic
+
+You will need to change the MQTT server your application connects to. {{% tts %}} has a new MQTT server address. You will also need to create API keys and update your MQTT credentials accordingly.
+
+[Read more about using the MQTT Server]({{< ref "/integrations/mqtt" >}}).
+
+{{% tts %}} also supports HTTP webhooks and PubSubs for nats.io or MQTT.
+
+### Payload Formats
+
+{{% ttnv2 %}} has support for payload decoders, converters, validators (for uplink) and encoders (for downlink). These can be either CayenneLP or Javascript functions.
+
+{{% tts %}} has support for an uplink payload formatter (similar to the payload decoder) and a downlink payload formatter (similar to the payload encoder). These can be set per application, and can even be overridden per end device.
+
+Migrating the {{% ttnv2 %}} payload encoder and decoder to an uplink and downlink payload formatter should be straightforward, since they have the same format.
```

Comment:
I think we need to link to a reference here, because we now also have some environment in addition to the payload and f_port arguments.

Reply:
Added a TODO comment with a reference to https://github.com/TheThingsNetwork/lorawan-stack/issues/598, since payload formatters are not yet documented.